docs/src/main/paradox/java/http/routing-dsl/httpApp.md

```diff
@@ -0,0 +1,69 @@
+<a id="http-app"></a>
+# HttpApp Bootstrap
+
+@@@ warning { title="API may change" }
+This is still experimental and the API is subjected to change in futures releases of Akka HTTP.
+@@@
+
+@@toc { depth=1 }
+
+## Introduction
+
+The objective of `HttpApp` is to reduce the boilerplate needed to start a server and handle its shutdown in a proper way.
+The current version of it, enables to create a new server just by extending `HttpApp` and implementing the `route()` method.
```

Comment:
I think I'd rather explain that this is a class to quickly get started, not talk about boilerplate at all, since with that "boilerplate" you get way more control and IMO in many prod apps you should be doing that (the more explicit way) anyway.